ui/ui-components/pages/_app.tsx

```diff
@@ -74,15 +79,30 @@ GrouparooNextApp.getInitialProps = async (appContext: AppContext) => {
   };
 
   try {
-    const navigationResponse: Actions.NavigationList = await client.request(
-      "get",
-      `/navigation`
-    );
+    let model: Models.GrouparooModelType | null = null;
+    let modelResponse: Actions.ModelView;
+    let navigationResponse: Actions.NavigationList;
+
+    const promises: Promise<any>[] = [
+      client.request<Actions.NavigationList>("get", `/navigation`),
+    ];
+
+    if (modelId) {
+      promises.push(
+        client.request<Actions.ModelView>("get", `/model/${modelId}`)
+      );
+    }
+
+    [navigationResponse, modelResponse] = await Promise.all(promises);
```

Comment:
```suggestion
    const [navigationResponse, modelResponse] = await Promise.all([
        client.request<Actions.NavigationList>("get", `/navigation`),
        modelId ? client.request<Actions.ModelView>("get", `/model/${modelId}`) : Promise.resolve()
    ]);
```
Something like this should work, and then you don't have a `Promise<any>[]`